packages/features/bookings/components/AvailableTimes.tsx

```diff
@@ -7,24 +11,162 @@ import { useLocale } from "@calcom/lib/hooks/useLocale";
 import { Button, SkeletonText } from "@calcom/ui";
 
 import { useBookerStore } from "../Booker/store";
+import { getQueryParam } from "../Booker/utils/query-param";
 import { useTimePreferences } from "../lib";
+import { useCheckOverlapWithOverlay } from "../lib/useCheckOverlapWithOverlay";
 import { SeatsAvailabilityText } from "./SeatsAvailabilityText";
 
+type TOnTimeSelect = (
+  time: string,
+  attendees: number,
+  seatsPerTimeSlot?: number | null,
+  bookingUid?: string
+) => void;
+
 type AvailableTimesProps = {
   slots: Slots[string];
-  onTimeSelect: (
-    time: string,
-    attendees: number,
-    seatsPerTimeSlot?: number | null,
-    bookingUid?: string
-  ) => void;
+  onTimeSelect: TOnTimeSelect;
   seatsPerTimeSlot?: number | null;
   showAvailableSeatsCount?: boolean | null;
   showTimeFormatToggle?: boolean;
   className?: string;
   selectedSlots?: string[];
 };
 
+const SlotItem = ({
+  slot,
+  seatsPerTimeSlot,
+  selectedSlots,
+  onTimeSelect,
+  showAvailableSeatsCount,
+}: {
+  slot: Slots[string][number];
+  seatsPerTimeSlot?: number | null;
+  selectedSlots?: string[];
+  onTimeSelect: TOnTimeSelect;
+  showAvailableSeatsCount?: boolean | null;
+}) => {
+  const { t } = useLocale();
+
+  const overlayCalendarToggled = getQueryParam("overlayCalendar") === "true";
+  const [timeFormat, timezone] = useTimePreferences((state) => [state.timeFormat, state.timezone]);
+  const bookingData = useBookerStore((state) => state.bookingData);
+  const layout = useBookerStore((state) => state.layout);
+  const hasTimeSlots = !!seatsPerTimeSlot;
+  const computedDateWithUsersTimezone = dayjs.utc(slot.time).tz(timezone);
+
+  const bookingFull = !!(hasTimeSlots && slot.attendees && slot.attendees >= seatsPerTimeSlot);
+  const isHalfFull = slot.attendees && seatsPerTimeSlot && slot.attendees / seatsPerTimeSlot >= 0.5;
+  const isNearlyFull = slot.attendees && seatsPerTimeSlot && slot.attendees / seatsPerTimeSlot >= 0.83;
+  const colorClass = isNearlyFull ? "bg-rose-600" : isHalfFull ? "bg-yellow-500" : "bg-emerald-400";
+
+  const { isOverlapping, overlappingTimeEnd, overlappingTimeStart } = useCheckOverlapWithOverlay(
+    computedDateWithUsersTimezone
+  );
+  const [overlapConfirm, setOverlapConfirm] = useState(false);
+
+  const onButtonClick = useCallback(() => {
+    if (!overlayCalendarToggled) {
+      onTimeSelect(slot.time, slot?.attendees || 0, seatsPerTimeSlot, slot.bookingUid);
+      return;
+    }
+    if (isOverlapping && overlapConfirm) {
+      setOverlapConfirm(false);
+      return;
+    }
+
+    if (isOverlapping && !overlapConfirm) {
+      setOverlapConfirm(true);
+      return;
+    }
+    if (!overlapConfirm) {
+      onTimeSelect(slot.time, slot?.attendees || 0, seatsPerTimeSlot, slot.bookingUid);
+    }
+  }, [
+    overlayCalendarToggled,
+    isOverlapping,
+    overlapConfirm,
+    onTimeSelect,
+    slot.time,
+    slot?.attendees,
+    slot.bookingUid,
+    seatsPerTimeSlot,
+  ]);
```

Comment:
Confusing callback that handles the onClick 

determines if the onClick should open the confirm action or if we should go straight to the next step